File: docs/site/samples/typescript/examples/guides_loader_memory.ts
/**
 * Loader guide — MemoryLoader: load raw text by key, then construct a Prompt.
 *
 * The kernel stays I/O-free; the loader is a separate, caller-invoked I/O leaf.
 */

import assert from "node:assert/strict";
import { test } from "node:test";
import { MemoryLoader, Prompt } from "prompting-press";

const GREET_YAML = `\
name: greet
role: user
body: "Hello {{ name }}"
variables:
  name: { type: string, trusted: true }
`;

test("MemoryLoader: load raw text then construct", async () => {
	const loader = new MemoryLoader({ greet: GREET_YAML });

	// load() returns raw text — parsing is a separate step.
	const raw = await loader.load("greet");
	const prompt = Prompt.fromYaml(raw);
	assert.equal(prompt.name, "greet");
});

File: docs/site/samples/python/examples/guides_loader_memory.py
# Copyright (C) 2024-2026 Sjors Robroek
# SPDX-License-Identifier: AGPL-3.0-only

"""Loader guide — MemoryLoader: load raw text by key, then construct a Prompt.

The kernel stays I/O-free; the loader is a separate, caller-invoked I/O leaf.
"""

from prompting_press import Prompt
from prompting_press.loader import MemoryLoader

GREET_YAML = """\
name: greet
role: user
body: "Hello {{ name }}"
variables:
  name: { type: string, trusted: true }
"""


def main() -> None:
    loader = MemoryLoader({"greet": GREET_YAML})

    # load() returns raw text — parsing is a separate step.
    raw = loader.load("greet")
    prompt = Prompt.from_yaml(raw)
    assert prompt.name == "greet"


if __name__ == "__main__":
    main()
